cel/canonical_eventlog_test.go

```diff
@@ -56,7 +59,120 @@ func TestCELEncodingDecoding(t *testing.T) {
 	if len(digestsMap[crypto.SHA1]) != 20 {
 		t.Errorf("SHA1 digest length doesn't match")
 	}
-	if reflect.DeepEqual(decodedcel.Records, (*cel).Records) == false {
+	if !reflect.DeepEqual(decodedcel.Records, cel.Records) {
 		t.Errorf("decoded CEL doesn't equal to the original one")
 	}
 }
+
+func TestCELMeasureAndReplay(t *testing.T) {
+	tpm := test.GetTPM(t)
+	defer client.CheckedClose(t, tpm)
+
+	cel := &CEL{}
+	measuredHashes := []crypto.Hash{crypto.SHA256, crypto.SHA1, crypto.SHA512}
+
+	someEvent := make([]byte, 10)
+	cosEvent := CosTlv{someEvent}
+	someEvent2 := make([]byte, 10)
+
+	rand.Read(someEvent2)
+	cosEvent2 := CosTlv{someEvent2}
+	appendOrFatal(t, cel, tpm, test.DebugPCR, measuredHashes, cosEvent)
+	appendOrFatal(t, cel, tpm, test.DebugPCR, measuredHashes, cosEvent2)
+
+	appendOrFatal(t, cel, tpm, test.ApplicationPCR, measuredHashes, cosEvent2)
+	appendOrFatal(t, cel, tpm, test.ApplicationPCR, measuredHashes, cosEvent)
+	appendOrFatal(t, cel, tpm, test.ApplicationPCR, measuredHashes, cosEvent)
+
+	replay(t, cel, tpm, measuredHashes,
+		[]int{test.DebugPCR, test.ApplicationPCR} /*shouldSucceed=*/, true)
```

Comment:
nit: should the inline comment locate after the comma?
[]int{test.DebugPCR, test.ApplicationPCR}, /*shouldSucceed=*/ true)

Reply:
I agree, however the VSCode golang formatter overwrites putting it after the ",". I can put it on the next line if it bothers you.

Reply:
Never mind, lets follow the formatter.

Reply:
nvm I just went ahead and put it on the next line.